crates/pagecache/src/config.rs

```diff
@@ -688,3 +697,20 @@ impl Config {
             .expect("should be able to truncate");
     }
 }
+
+/// See the Kernel's documentation for more information about this subsystem, found at:
+///  [Documentation/cgroup-v1/memory.txt](https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt)
+#[cfg(target_os="linux")]
+fn get_cgroup_memory_limit() -> u64 {
+    std::fs::File::open("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+        .and_then(read_u64_from)
+        .unwrap_or(0)
+}
```

Comment:
In my mind I imagined a more annoying implementation than this, but this looks nice! The one change I would make to this is to return an Option (by replacing `.unwrap_or(0)` with `.ok()` to convert the result to an Option, and then in the caller above instead of checking for if it's above 0, using an `if let Some(mem) = get_cgroup_memory_limit() { ...`

Reply:
> I imagined a more annoying implementation 

What do you think of? I can abstract it more, reuse `cgroup-rs` or fallback to checking rlimit/as?

Reply:
I think the cgroup stuff itself is good!